app/src/androidTest/java/com/github/fribourgsdp/radio/PlayerGameHandlerTest.kt

```diff
@@ -33,6 +34,15 @@ class PlayerGameHandlerTest {
         `when`(mockSnapshot.getString("current_song")).thenReturn(null)
         `when`(mockSnapshot.get("scores")).thenReturn(scores)
         `when`(mockSnapshot.getBoolean("finished")).thenReturn(false)
+        mockSnapshot2 = mock(DocumentSnapshot::class.java)
+        `when`(mockSnapshot2.getString("singer")).thenReturn("")
+        `when`(mockSnapshot2.exists()).thenReturn(true)
+        `when`(mockSnapshot2.getLong("current_round")).thenReturn(round)
+        `when`(mockSnapshot2.get("song_choices")).thenReturn(listOfSongs)
+        `when`(mockSnapshot2.get("song_choices_lyrics")).thenReturn(hashMapOf("Momentum" to "Lorem Ipsum"))
+        `when`(mockSnapshot2.getString("current_song")).thenReturn("Momentum")
+        `when`(mockSnapshot2.get("scores")).thenReturn(scores)
+        `when`(mockSnapshot2.getBoolean("finished")).thenReturn(false)
```

Comment:
Since you only use `mocksnapshot2` once you could simply write this on `mocksnapshot` before using this in your test. That way this isn't computed before every test and only used once.